dwave/system/temperatures.py

```diff
@@ -249,373 +391,736 @@ def maximum_pseudolikelihood_temperature(bqm=None,
+                warnings.simplefilter(action="ignore", category=RuntimeWarning)
+                expFactor = np.exp(
+                    np.sum(en1 * x[:, np.newaxis, np.newaxis], axis=0)
+                )
+                norm = expFactor + 2 + 1 / expFactor
+            return -np.sum(
+                [
+                    [
+                        sample_weights
+                        * np.sum(en1[i, :, :] * en1[j, :, :] / norm, axis=1)
+                        for i in range(en1.shape[0])
+                    ]
+                    for j in range(en1.shape[0])
+                ],
+                axis=-1,
+            )
+
+    return dd_mean_log_pseudo_likelihood
+
+
+def maximum_pseudolikelihood(
+    en1: Optional[np.ndarray] = None,
+    bqms: Union[None, List[dimod.BinaryQuadraticModel]] = None,
+    sampleset: Union[None, dimod.SampleSet, Tuple[np.ndarray, List]] = None,
+    num_bootstrap_samples: int = 0,
+    seed: Optional[int] = None,
+    x0: Union[None, List, np.ndarray] = None,
+    optimize_method: Optional[str] = None,
+    kwargs_opt: Optional[dict] = None,
+    sample_weights: Optional[np.ndarray] = None,
+    return_optimize_object: bool = False,
+    degenerate_fields: Optional[bool] = None,
+    use_jacobian: bool = True,
+) -> Tuple:
+    """Maximimum pseudolikelihood estimator for exponential models.
+
+    Uses the SciPy optimize method to solve the maximum pseudolikelihood problem
+    of weight estimation for an exponential model with exponent defined by a
+    weighted sum of bqms.
+    Assuming the Probability of states s is given by an exponential model
+    :math:`P(s)=1/Z exp(- x_i H_i(s))` for a given list of functions (bqms)
+    estimate x. The exponential model is also called a Boltzmann distribution.
+    Code is designed assuming the bqms are dense (a function of all or most of
+    the variables), although technically operation although sparse bqms
+    like :math:`H(s)=h_i s_i` or :math:`H(s)=J_{ij}s_i s_j` are technically allowed.
+
+    Note common reasons for parameter inference failure include:
+    - Too few samples, insufficient to resolve parameters
+    - The samplest is singular or insensitive with respect to some parameter
+      (e.g. local minima or plateus only).
+    - bqms are too closely related, or weakly dependent on the sampleset
+      variability (e.g. nearly collinear).
+
+    Args:
+        bqms: A list of Binary quadratic models [H_i] describing the sample
+            distribution as :math:`P(s) ~ exp(sum_i x_i H_i(s))` for unknown
+            model parameters x.
+        sampleset: A set of samples, as a dimod Sampleset or samples-like object.
+        en1: Effective fields as an np.ndarray (site labels not required).
+            Derived from the ``bqms`` and ``sampleset`` if not provided.
+            First dimension indexes samples, second dimension indexes sites.
+            Ordering doesn't matter, but should be consistent with sample_weights.
```

Comment:
Just moving `en1` up to first since that's the order in the signature.
```suggestion
        en1: Effective fields as an np.ndarray (site labels not required).
            Derived from the ``bqms`` and ``sampleset`` if not provided.
            First dimension indexes samples, second dimension indexes sites.
            Ordering doesn't matter, but should be consistent with sample_weights.
        bqms: A list of Binary quadratic models [H_i] describing the sample
            distribution as :math:`P(s) ~ exp(sum_i x_i H_i(s))` for unknown
            model parameters x.
        sampleset: A set of samples, as a dimod Sampleset or samples-like object.
```